app/[locale]/[slug]/page.tsx

```diff
@@ -1,18 +1,21 @@
 import { Content, ErrorState } from '@/components'
 import { SITE } from '@/constants'
-import { getAllPages, getSinglePage } from '@/lib'
+import { getSinglePage } from '@/lib'
 import { Heading, Section } from '@/components/ui'
 import { Metadata } from 'next'
+import { getAllPages } from '@/lib/api'
 
 export const revalidate = 60
 export const dynamicParams = true
 
 export async function generateStaticParams() {
   const pages = await getAllPages()
-  return pages.data.map((page) => ({
-    id: String(page.id),
-    slug: String(page.slug),
-  }))
+  return (
+    pages.data?.map((page) => ({
```

Comment:
QQ: Is there a specific reason for that approach? If the data is missing, perhaps it would be more appropriate to throw an error rather than bypassing it.

Reply:
I tried to keep the same original code with minimum changes. Most of the application checks for errors, but it is not here. For the docs about generateStaticSomething, we should return an empty array. I'll make it return a data with an empty array in case of error.

Reply:
https://linear.app/brinca/issue/BRINCA-86/make-api-return-data-as-an-array-for-all-cases